From b5d74a8c8803b276ff1765e294786b010a702859 Mon Sep 17 00:00:00 2001
From: Emmanuel Berthier <emmanuel.berthier@intel.com>
Date: Tue, 28 Jun 2016 14:57:57 +0200
Subject: [PATCH] [VMSCAN] Correct a missing initialization in
 debug_shrinker_show()

sc.nid is not initialized and generates crash:

[   27.281852]  [<ffffffff81a4dd0f>] _raw_spin_lock+0x1f/0x30
[   27.288007]  [<ffffffff8118771e>] list_lru_count_node+0x1e/0x70
[   27.294648]  [<ffffffff811bc299>] super_cache_count+0x59/0xf0
[   27.301095]  [<ffffffff811720b5>] debug_shrinker_show+0x75/0xa0
[   27.307736]  [<ffffffff811de950>] seq_read+0xe0/0x3e0
[   27.313405]  [<ffffffff811b9d3b>] vfs_read+0x9b/0x160
[   27.319074]  [<ffffffff811ba809>] SyS_read+0x49/0xb0
[   27.324647]  [<ffffffff81a564b2>] system_call_fastpath+0x16/0x1b

Change-Id: Ice7d80c87ce6555801ce234ee24c07f3fe791df8
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-33141
Signed-off-by: Emmanuel Berthier <emmanuel.berthier@intel.com>
Reviewed-on: https://android.intel.com:443/519949
---
 mm/vmscan.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/vmscan.c b/mm/vmscan.c
index 039ed2c..ae3ab8c 100644
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -185,6 +185,7 @@ static int debug_shrinker_show(struct seq_file *s, void *unused)
 
 	sc.gfp_mask = -1;
 	sc.nr_to_scan = 0;
+	sc.nid = 0;
 
 	down_read(&shrinker_rwsem);
 	list_for_each_entry(shrinker, &shrinker_list, list) {
-- 
1.9.1

